src/js/background.js

```diff
@@ -0,0 +1,214 @@
+const setLoadingState = async (active, tabId) => {
+  const [calExist] = await browser.tabs.executeScript(tabId, {
+    code: `document.querySelector(".cal-loading") !== null;`,
+  });
+
+  if (!calExist) {
+    await browser.tabs.executeScript(tabId, {
+      file: "/src/js/createLoading.js",
+    });
+    await browser.tabs.insertCSS(tabId, { file: "/src/css/createLoading.css" });
+  }
```

Comment:
Is there a chance that the script and CSS file won't be loaded? I see they are mentioned as `content_scripts` in the `manifest.json` and according to [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts) they are loaded automatically. Perhaps you need to specify [`run_at`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts#run_at) for them. Relying on `content_scripts` loading seems more declarative and simpler than manually injecting them